Update

L1ghtmann opened the pull request:
Resolves #6 and resolves #7

Changes
- Updates workflow
- Adds aarch64, RHEL, and Arch builds
- Adds logic to confirm linux_version is one of the supported versions
- Removes borked plutil

Tested successfully [here](https://github.com/L1ghtmann/swift-toolchain-linux/actions/runs/9341624422)
- AArch64 builds failed as they are absent from my DT release currently (see the [separate DT pr](https://github.com/kabiroberai/darwin-tools-linux/pull/7))

L1ghtmann committed:
Update actions + copyright

L1ghtmann committed:
Test

L1ghtmann committed:
Fix

L1ghtmann committed:
Fix

L1ghtmann committed:
Swift v5.9.2 & remove c++ headers

L1ghtmann committed:
Rm plutil

L1ghtmann committed:
Add RHEL support

L1ghtmann committed:
I'm a genius

L1ghtmann committed:
Add supported plat check + update readme

L1ghtmann committed:
Support ubi9-aarch64

L1ghtmann committed:
Done testing

kabiroberai commented on create-toolchain:
```diff
@@ -106,11 +112,26 @@ cp -a "${mac_toolchain}"/usr/include linux/iphone/include
-dl "darwin-tools-${linux_version}.tar.gz" "https://github.com/kabiroberai/darwin-tools-linux/releases/download/v${DARWIN_TOOLS_VERSION}/darwin-tools-${linux_version}.tar.gz"
-tar -xzvf "darwin-tools-${linux_version}.tar.gz"
+# If distro isn't debian-based, use ubuntu20.04 as darwintools base
+if ! [[ "${linux_version}" == *"ubuntu"* ]]; then
```
readability nit: let's invert this statement to `if [[ "${linux_version}" == *"ubuntu"* ]]; then` and swap the branches around.

kabiroberai reviewed (commented):
Thanks for this! Just one nit.

L1ghtmann committed:
Requested change

L1ghtmann commented:
We may want to investigate the `dsymutil` issue or remove it from the darwin-tools toolchain outright as it prevents Theos builds from completing successfully on Arch/RHEL. When removed, the build proceeds as expected just without 'generating debug symbols'.

L1ghtmann committed:
Revert to Swift v5.8.1 for ABI reasons

L1ghtmann committed:
Add Arch support, adjust script args, update downloads

L1ghtmann commented:
This has been updated to support recent changes to the [accompanying DT pr](https://github.com/kabiroberai/darwin-tools-linux/pull/7). Converted to draft in the interim while we decide what to do regarding the two failing DT builds.

L1ghtmann committed:
Update

L1ghtmann committed:
Fix paths for fixup

L1ghtmann committed:
Bump swift vers and re-add aarch64 ubuntu vers to matrix

L1ghtmann committed:
There's an iphone one too

L1ghtmann commented:
Pr updated. Changes from upstream include:
- Updates to CI actions
- Addition of 24.04, fedora39, and archlinux builds
- Swift bumped to v6.0.2
- Updated script to account for args passed and supported plats
- Explicitly remove `plutil` from bin as it's been reported borked a number of times. Will now include `plistutil` instead
- Generalize untar commands 

Removed earlier additions of aarch64 fedora and arch builds per my comments in the associated pr.

kabiroberai commented on .github/workflows/release.yml:
```diff
@@ -24,15 +24,16 @@ jobs:
       matrix:
-        os: [ubuntu22.04-aarch64, ubuntu22.04, ubuntu20.04-aarch64, ubuntu20.04, ubuntu18.04]
-    runs-on: macos-13
+        os: [ubuntu24.04, ubuntu24.04-aarch64, ubuntu22.04, ubuntu22.04-aarch64, ubuntu20.04, ubuntu20.04-aarch64, fedora39, archlinux]
```
can we include `fedora39-aarch64` and `archlinux-aarch64` here as well? I'll upload those manually to the darwin-tools repo

kabiroberai commented on .github/workflows/release.yml:
```diff
@@ -24,15 +24,16 @@ jobs:
-        os: [ubuntu22.04-aarch64, ubuntu22.04, ubuntu20.04-aarch64, ubuntu20.04, ubuntu18.04]
-    runs-on: macos-13
+        os: [ubuntu24.04, ubuntu24.04-aarch64, ubuntu22.04, ubuntu22.04-aarch64, ubuntu20.04, ubuntu20.04-aarch64, fedora39, archlinux]
+    runs-on: macos-14
```
let's bump this to `macos-15` and further update the SDK to Xcode 16.2?

kabiroberai commented on create-toolchain:
```diff
@@ -6,13 +6,16 @@ DARWIN_TOOLS_VERSION="2.2.1"
+	fi
+else
+	if [[ $# -ne 2 ]]; then
+		echo "Usage: $0 <linux version (e.g., ubuntu20.04)> [swift version (e.g., 5.8.1)]"
```
nit: square brackets conventionally denote optional arguments so let's use angled brackets here
```suggestion
		echo "Usage: $0 <linux version (e.g., ubuntu20.04)> <swift version (e.g., 6.0.3)>"
```

L1ghtmann replied:
TIL! Sure thing

kabiroberai reviewed (commented):
LGTM, just a few nits/requests!

L1ghtmann committed:
Requested changes

kabiroberai commented on create-toolchain:
```diff
@@ -106,11 +111,13 @@ cp -a "${mac_toolchain}"/usr/include linux/iphone/include
 
 info "Installing Darwin tools"
 # The hierarchy lines up with ours just right so we dont need to -C
 dl "darwin-tools-${linux_version}.tar.gz" "https://github.com/kabiroberai/darwin-tools-linux/releases/download/v${DARWIN_TOOLS_VERSION}/darwin-tools-${linux_version}.tar.gz"
```
let's update `DARWIN_TOOLS_VERSION` before merging this (since we need your changes for plistutil). I'll cut a DT release once CI passes and I have the chance to test it.

L1ghtmann committed:
Bump DT ver
